types/testingutils/share_map.go

```diff
@@ -17,20 +17,31 @@ func ValidatorIndexList(limit int) []int {
 	return ret
 }
 
-func KeySetMapForValidatorIndexList(valIndexes []int) map[phase0.ValidatorIndex]*TestKeySet {
+func KeySetMapForValidatorIndexList(valIndexes []phase0.ValidatorIndex) map[phase0.ValidatorIndex]*TestKeySet {
 	ret := make(map[phase0.ValidatorIndex]*TestKeySet)
 	for _, valIdx := range valIndexes {
-		ks, exists := TestingKeySetMap[phase0.ValidatorIndex(valIdx)]
+		ks, exists := TestingKeySetMap[valIdx]
 		if !exists {
 			panic(fmt.Sprintf("Validator index %v does not exist in TestingKeySetMap", valIdx))
 		}
-		ret[phase0.ValidatorIndex(valIdx)] = ks
+		ret[valIdx] = ks
 	}
 	return ret
 }
 
 func KeySetMapForValidators(limit int) map[phase0.ValidatorIndex]*TestKeySet {
-	return KeySetMapForValidatorIndexList(ValidatorIndexList(limit))
+	if limit <= 0 {
+		return map[phase0.ValidatorIndex]*TestKeySet{}
+	}
+	validators := make([]phase0.ValidatorIndex, limit)
+	for i := 0; i < limit; i++ {
+		validatorIndex := (i + 1)
+		if validatorIndex < 0 {
+			panic("Invalid validator index")
+		}
+		validators[i] = phase0.ValidatorIndex(validatorIndex)
+	}
+	return KeySetMapForValidatorIndexList(validators)
```

Comment:
why not change `ValidatorIndexList`?

Reply:
It would incur changes through many tests. I can definitely do it here if you want. But, since it's not the goal of this PR, I decided to leave this for a future PR.